backend/src/middleware/permissionsMiddleware.ts

```diff
@@ -370,11 +372,28 @@ const noEmailFilter = rule({
 
 const publicRegistration = rule()(() => CONFIG.PUBLIC_REGISTRATION)
 
-const inviteRegistration = rule()(async (_parent, args, { _user, driver }) => {
+const inviteRegistration = rule()(async (_parent, args, context: Context) => {
   if (!CONFIG.INVITE_REGISTRATION) return false
   const { inviteCode } = args
-  const session = driver.session()
-  return validateInviteCode(session, inviteCode)
+  return validateInviteCode(context, inviteCode)
+})
+
+const isAllowedToGenerateGroupInviteCode = rule({
+  cache: 'no_cache',
+})(async (_parent, args, context: Context) => {
+  if (!context.user) return false
+
+  return !!(
+    await context.database.query({
+      query: `
+    MATCH (user:User{id: user.id})-[membership:MEMBER_OF]->(group:Group {id: $args.groupId})
+    WHERE (group.type IN ['closed','hidden'] AND membership.role IN ['admin', 'owner'])
+      OR (NOT group.type IN ['closed','hidden'] AND NOT membership.role = 'pending')
```

Comment:
How can you be a pending member of a public group?

Reply:
When an admin assigns you this role